File: app/src/main/java/io/ipoli/android/app/api/UrlProvider.java
package io.ipoli.android.app.api;

import java.net.MalformedURLException;
import java.net.URL;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 3/30/17.
 */

public interface UrlProvider {
    URL sync();

    URL api();

    URL createUser();

    URL migrateUser(String firebasePlayerId);

    static URL getURL(String path) {
        try {
            return new URL(path);
        } catch (MalformedURLException e) {
            return null;
        }
    }
}

File: app/src/main/java/io/ipoli/android/common/api/UrlProvider.kt
package io.ipoli.android.common.api

import io.ipoli.android.ApiConstants
import java.net.URL

/**
 * Created by Polina Zhelyazkova <polina@mypoli.fun>
 * on 3/23/18.
 */
interface UrlProvider {
    val getMembershipStatus: URL
    val migratePlayer: URL
}

class ProdUrlProvider : UrlProvider {
    override val getMembershipStatus = URL("${ApiConstants.PROD_API_URL}subscriptions/")
    override val migratePlayer = URL("${ApiConstants.PROD_API_URL}migrations/")
}

class DevUrlProvider : UrlProvider {
    override val getMembershipStatus = URL("${ApiConstants.DEV_API_URL}subscriptions/")
    override val migratePlayer = URL("${ApiConstants.DEV_API_URL}migrations/")
}